internal/state/conditions/httproute.go

```diff
@@ -0,0 +1,72 @@
+package conditions
+
+import (
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// RouteCondition defines a condition to be reported in the status of an HTTPRoute.
+type RouteCondition struct {
+	Type    v1beta1.RouteConditionType
+	Status  metav1.ConditionStatus
+	Reason  v1beta1.RouteConditionReason
+	Message string
+}
+
+// DeduplicateRouteConditions removes duplicate conditions based on the condition type.
+// The last condition wins.
+func DeduplicateRouteConditions(conds []RouteCondition) []RouteCondition {
+	uniqueConds := make(map[v1beta1.RouteConditionType]RouteCondition)
```

Comment:
Shouldn't this be initialized to `len(conds)`?

Reply:
There aren't many condition Types - right now we support the Gateway API `Accepted` and `TODO` (homegrown).
I expect we will have duplications in `conds` slice but the number of condition types will stay low. With that in mind, I don't think it makes sense to consider any optimizations with map initialization here.